File: Tests/Tests/HistoryPageTests.cs
using System.Collections.Generic;
using Atata;
using NUnit.Framework;

namespace Tests.Tests
{
    public class HistoryPageTests : BaseTest
    {
        [Test]
        public void ClearHistoryTest() =>
            OpenDepositPage()
                .Populate("1000", "100", "300").Calculate()
                .Populate("1500", "25", "100").Calculate()
                .OpenHistory().Clear().ReturnToCalculator().OpenHistory()
                .History.Headers.Should.ContainHavingContent(TermMatch.Equals, Defaults.HistoryHeaders)
                .History.Rows.Count.Should.Equal(1);

        [Test]
        public void AssertHistoryTableTest() =>
            OpenDepositPage()
                .Populate("1000", "100", "300").Calculate()
                .RetrieveData(out var row1)
                .Populate("1500", "25", "100").Calculate()
                .RetrieveData(out var row2)
            .OpenHistory()
                .History.Headers.Should.ContainHavingContent(TermMatch.Equals, Defaults.HistoryHeaders)
                .History.Rows.Count.Should.Equal(3)
                .History.Rows[1].Content.Should.Equal(row2)
                .History.Rows[2].Content.Should.Equal(row1);

        [Test]
        public void HistoryShowsLast9RecordsTest()
        {
            // Arrange
            var records = new List<string>{""};
            var depositPage = OpenDepositPage();

            // Act
            for (var i = 1; i < 12; i++)
            {
                depositPage.Populate("1000", i.ToString(), "300").Calculate().RetrieveData(out var data);
                records.Insert(1, data);
            }
            var history = depositPage.OpenHistory().History;

            // Assert
            history.Rows.Count.Should.Equal(9+1);

            for (var i = 1; i < history.Rows.Count; i++)
            {
                history.Rows[i].Content.Should.Equal(records[i]);
            }
        }

        [TestCase("$ - US dollar", "123.456.789,00", "dd-MM-yyyy")]
        [TestCase("€ - euro", "123 456 789.00", "MM/dd/yyyy")]
        [TestCase("£ - Great Britain Pound", "123 456 789,00", "MM dd yyyy")]
        public void HistoryRespectSettingsTest(string currency, string numberFormat, string dateFormat) =>
            OpenDepositPage()
                .Populate("100000", "99", "299").Calculate()
                .RetrieveData(out var row1, dateFormat, numberFormat)
                .Populate("100000", "75", "299").Calculate()
                .RetrieveData(out var row2, dateFormat, numberFormat)
            .OpenSettings().Set(currency, numberFormat, dateFormat)
            .OpenHistory()
                .History.Headers.Should.ContainHavingContent(TermMatch.Equals, Defaults.HistoryHeaders)
                .History.Rows.Count.Should.Equal(3)
                .History.Rows[1].Content.Should.Equal(row2)
                .History.Rows[2].Content.Should.Equal(row1);
    }
}

File: Tests/src/tests/HistoryPageTests.java
package tests;

import java.io.IOException;
import java.text.ParseException;
import java.util.*;
import java.util.stream.Collectors;
import java.util.stream.Stream;

import org.junit.Assert;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import utilities.Users;

public class HistoryPageTests extends BaseTest {
	@BeforeEach
	public void openDepositPage() throws IOException {
		String user = "settings";

		Users.delete(user);
		Users.register(user);

		depositPage = loginPage.login(user);
	}
	
	@Test
	public void clearHistoryTest() {
		depositPage.populate("1000", "10", "299").calculate();
		depositPage.populate("1250", "25", "199").calculate();
		
		historyPage = depositPage
			.openHistory()
			.clear()
			.openCalculator()
			.openHistory();
		
		Assert.assertEquals(0, historyPage.getHistorySize());
	}
	
	@Test
	public void assertHistoryTest() throws ParseException {
		List<String> first  = depositPage.populate("1000", "10", "299").calculate().getData();
		List<String> second = depositPage.populate("1250", "25", "199").calculate().getData();
		
		historyPage = depositPage.openHistory();
		
		Assertions.assertAll(
			() -> Assert.assertEquals(2,      historyPage.getHistorySize()),	
			() -> Assert.assertEquals(first,  historyPage.getTableRow(2)),
			() -> Assert.assertEquals(second, historyPage.getTableRow(1))
		);
	}

	@Test
	public void historyShowsLast9RecordsTest() throws ParseException
    {
        var expectedHistory = new ArrayList<List<String>>();

        for (var i = 1; i < 12; i++)
        {
    		List<String> row = depositPage.populate("1000", String.valueOf(i), "299").calculate().getData();
    		expectedHistory.add(0, row);
        }
        
        historyPage = depositPage.openHistory();

        expectedHistory = (ArrayList<List<String>>) expectedHistory.stream().limit(9).collect(Collectors.toList());
        
        Assert.assertEquals(expectedHistory, historyPage.getHistory());
    }

	static Stream<Arguments> settings(){
	    return Stream.of(
	       Arguments.of("$ - US dollar", "123.456.789,00", "dd-MM-yyyy"),
	       Arguments.of("€ - euro", "123 456 789.00", "MM/dd/yyyy"),
	       Arguments.of("£ - Great Britain Pound", "123 456 789,00", "MM dd yyyy")
	    );
	}

	@ParameterizedTest
	@MethodSource("settings")
	public void historyRespectSettingsTest(String currency, String numberFormat, String dateFormat) throws ParseException {
		depositPage.openSettings().set(dateFormat, numberFormat, currency).save();
		
		List<String> first  = depositPage.populate("100000", "99", "299").calculate().getData(dateFormat, numberFormat);
		List<String> second = depositPage.populate("100000", "75", "299").calculate().getData(dateFormat, numberFormat);

		historyPage = depositPage.openHistory();
		
		Assertions.assertAll(
			() -> Assert.assertEquals(2,      historyPage.getHistorySize()),	
			() -> Assert.assertEquals(first,  historyPage.getTableRow(2)),
			() -> Assert.assertEquals(second, historyPage.getTableRow(1))
		);
	}
}